tests/vmi_lifecycle_test.go

```diff
@@ -569,6 +569,92 @@ var _ = Describe("[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:compon
+				pod := renderPkillAllPod("kubelet")
+				pod.Spec.NodeName = nodeName
+				_, err = virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Create(pod)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("starting another VMI on the same node, to verify kubelet is running again")
+				newVMI := newCirrosVMI()
+				newVMI.Spec.NodeSelector = map[string]string{"kubernetes.io/hostname": nodeName}
+				Eventually(func() error {
+					newVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(newVMI)
+					Expect(err).To(BeNil())
+					return nil
+				}, 100, 10).Should(Succeed(), "Should be able to start a new VM")
+
+				By("checking if the VMI with bridged networking is still running, it will verify the CNI didn't cause the pod to be killed")
+				tests.WaitForSuccessfulVMIStart(bridgeVMI)
+			})
+
+			It("[test_id:5276]VMIs with Bridge Networking should work with Duplicate Address Detection (DAD)", func() {
+				VerifyUserDataVMI := func(vmi *v1.VirtualMachineInstance, commands []expect.Batcher, timeout time.Duration) {
+					By("Checking that the VirtualMachineInstance serial console output equals to expected one")
+					Expect(console.SafeExpectBatch(vmi, commands, int(timeout.Seconds()))).To(Succeed())
+				}
+
+				userData := fmt.Sprintf("#cloud-config\npassword: %s\nchpasswd: { expire: False }\n", fedoraPassword)
+
+				bridgeVMI := vmi
+				// Remove the masquerade interface to use the default bridge one
+				bridgeVMI.Spec.Domain.Devices.Interfaces = nil
+				bridgeVMI.Spec.Networks = nil
+				v1.SetDefaults_NetworkInterface(bridgeVMI)
+				Expect(bridgeVMI.Spec.Domain.Devices.Interfaces).NotTo(BeEmpty())
+
+				By("starting a VMI with bridged network on a node")
+				bridgeVMI = tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), userData)
+				//bridgeVMI, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(bridgeVMI)
+				//Expect(err).To(BeNil(), "Should submit VMI successfully")
+
+				// Start a VirtualMachineInstance with bridged networking
+
+				By("Starting a VirtualMachineInstance")
+				obj, err := virtClient.RestClient().Post().Resource("virtualmachineinstances").Namespace(tests.NamespaceTestDefault).Body(bridgeVMI).Do().Get()
+				Expect(err).To(BeNil())
+				By("Waiting the VirtualMachineInstance start")
+				bridgeVMI, ok := obj.(*v1.VirtualMachineInstance)
+				Expect(ok).To(BeTrue(), "Object is not of type *v1.VirtualMachineInstance")
+				tests.WaitForSuccessfulVMIStart(bridgeVMI)
+				tests.VerifyDummyNicForBridgeNetwork(bridgeVMI)
+
+				// Eventually(func() []v1.VirtualMachineInstanceNetworkInterface {
+				// 	return bridgeVMI.Status.Interfaces
+				// }, 100, 10).Should(HaveLen(1), "VM should have an interface")
+
+				// eth0IP := bridgeVMI.Status.Interfaces[0].IP
+				dadTest := "/usr/sbin/arping -D -I eth0 -c 2 $(hostname -i) | grep Received | cut -d ' ' -f 2"
+				VerifyUserDataVMI(vmi, []expect.Batcher{
```

Comment:
Instead of this, can you use our helpers? Should look somehow like this:

```go
			Expect(console.LoginToFedora(vmi)).To(Succeed())
			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
				&expect.BSnd{S: "/usr/sbin/arping -D -I eth0 -c 2 $(hostname -i) | grep Received | cut -d ' ' -f 2\n"},
				&expect.BExp{R: "0"},
			}, 200)).To(Succeed())
		}
```

Reply:
+1 to Roman's suggestion.
Adding another option -
You can use `tests.WaitUntilVMIReady(bridgedVmi, console.LoginToFedora)` instead of `tests.WaitForSuccessfulVMIStart(bridgeVMI)`

And -
`Expect(console.RunCommand(bridgedVmi, dadTest, 200*time.Second))` instead of `VerifyUserDataVMI`.